src/avlizer_confluent.erl

```diff
@@ -351,30 +359,30 @@ unify_ref({Name, Fp}) when is_list(Fp) ->
   unify_ref({Name, iolist_to_binary(Fp)});
 unify_ref(Ref) -> Ref.
 
--spec do_download(string(), ref()) -> {ok, binary()} | {error, any()}.
-do_download(RegistryURL, RegId) when is_integer(RegId) ->
-  URL = RegistryURL ++ "/schemas/ids/" ++ integer_to_list(RegId),
-  httpc_download(URL);
-do_download(RegistryURL, {Name, Fp}) ->
+-spec do_download({string(), string()}, ref()) -> {ok, binary()} | {error, any()}.
```

Comment:
headers is a list not a string.
maybe delete this spec since it's internal.

Reply:
Good catch @zmstone